services/brig/src/Brig/Sem/Now/MonadIO.hs

```diff
@@ -0,0 +1,41 @@
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2022 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+
+module Brig.Sem.Now.MonadIO (brigTimeToMonadIO) where
+
+import Data.Time
+import Imports
+import Polysemy
+import Wire.Sem.FromUTC
+import Wire.Sem.Now
+
+-- | An interpreter of the 'Now' effect to MonadIO.
+brigTimeToMonadIO ::
+  forall m r a.
+  (MonadIO m, Member (Embed m) r) =>
+  IO UTCTime ->
+  Sem (Now ': r) a ->
+  Sem r a
+brigTimeToMonadIO ioTime = interpret now
+  where
+    -- If this helper was inlined, GHC wouldn't be able to figure out
```

Comment:
I think it should if you enable `{-# OPTIONS_GHC -fplugin=Polysemy.Plugin #-}`

Reply:
You'll have to show me in the next pairing session how to make use of the plugin.